src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonBuildIntegratedNuGetProject.cs

```diff
@@ -308,12 +318,26 @@ private JObject GetJson()
                     string.Format(Strings.ErrorLoadingPackagesConfig, _jsonConfig.FullName, ex.Message), ex);
             }
         }
+
         private async Task SaveJsonAsync(JObject json)
         {
             using (var writer = new StreamWriter(_jsonConfig.FullName, false, Encoding.UTF8))
             {
                 await writer.WriteAsync(json.ToString());
             }
         }
+
+        private void UpdateFramework(JObject json)
+        {
+            var frameworks = JsonConfigUtility.GetFrameworks(json);
+            if (frameworks.Count() == 1
+                && IsUAPFramework(frameworks.First())
```

Comment:
Had an offline chat with Alex, decided to add a check on the `newTargetFramework` for UAP and get input from PM on the need for check for json framework.